backend/src/common/plugins/auth-plugin.ts

```diff
@@ -0,0 +1,51 @@
+import fastifyPlugin from 'fastify-plugin';
+import { AuthApiPath, UserValidationMessage } from 'shared';
+
+import { PluginNames } from '~/common/enums/enums.js';
+import { extractTokenFromHeaders } from '~/common/helpers/helpers.js';
+import { HttpCode, HttpError } from '~/common/http/http.js';
+
+import { type AuthOptions } from './types/types.js';
+
+const authPlugin = fastifyPlugin(
+    (fastify, { jwtService, apis }: AuthOptions, done) => {
+        fastify.decorateRequest('user', null);
+
+        const routes = apis.flatMap((api) =>
+            api.routes.map((element) => element.path),
+        );
+        const excludedRoutes = [AuthApiPath.SIGN_IN, AuthApiPath.SIGN_UP];
+        const filteredRoutes = routes.filter((route) =>
+            excludedRoutes.some((excRoute) => route.includes(excRoute)),
+        );
```

Comment:
This approach won't be sufficient. It's important to consider the ApiPath. 
e.g. using :id for one ApiPath will automatically whitelist all :id enpoints.

Additionally,
e.g. request.routeOptions.url -> /api/v1/users/
And instead of finding matches anywhere in the URL, would be nice to remove the /api/v1 part maybe with regular expression and then check for an exact match.
This also should eliminate the need to pass `apis` as a parameter.